source/agora/flash/OnionPacket.d

```diff
@@ -67,11 +68,10 @@ public struct OnionPacket
     // shared secret for decrypting
     public Point ephemeral_pk;
 
-    // encrypted payload, decrypts to a `Payload`
-    public EncryptedPayload encrypted_payload;
-
-    // todo: replace with actual HMAC
-    public Hash hmac;
+    // encrypted payload. The first one decrypts & deserializes to a `Payload`.
+    // the next payload may or may not be legitimate, but it depends entirely
+    // on the next node's ability to decrypt the payload.
+    public EncryptedPayload[20] encrypted_payloads;
```

Comment:
Does this limit our hop count now?

Reply:
Sure. Although we could technically set this to a higher limit if we want to.